addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/handler/MySensorsThingHandler.java

```diff
@@ -0,0 +1,396 @@
+
+        Bridge bridge = getBridge();
+        myBridgeHandler = (MySensorsBridgeHandler) bridge.getHandler();
+
+        return myBridgeHandler;
+    }
+
+    private void handleChildUpdateEvent(MySensorsVariable var) {
+        String channelName = getChannelNameFromVar(var);
+        State newState = loadAdapterForChannelType(channelName).stateFromChannel(var);
+        logger.debug("Updating channel: {}({}) value to: {}", channelName, var.getType(), newState);
+        updateState(channelName, newState);
+
+    }
+
+    private void handleBatteryUpdateEvent(MySensorsNode node) {
+        logger.debug("Updating channel: {} value to: {}", CHANNEL_BATTERY, node.getBatteryPercent());
+        updateState(CHANNEL_BATTERY, new DecimalType(node.getBatteryPercent()));
+    }
+
+    private String getChannelNameFromVar(MySensorsVariable var) {
+        // Cover thing has a specific behavior
+        if (getThing().getThingTypeUID().equals(MySensorsBindingConstants.THING_TYPE_COVER)) {
+            return MySensorsBindingConstants.CHANNEL_COVER;
+        } else {
+            return CHANNEL_MAP.get(var.getType());
+        }
+    }
+
+    private MySensorsTypeConverter loadAdapterForChannelType(String channelName) {
+        return TYPE_MAP.get(channelName);
+    }
+
+    /**
+     * If a new message is received via the MySensors bridge it is handed over to the ThingHandler
+     * and processed in this method. After parsing the message the corresponding channel is updated.
+     *
+     * @param msg The message that was received by the MySensors gateway.
+     */
+    private void handleIncomingMessageEvent(MySensorsMessage msg) {
+        // Am I the all knowing node that receives all messages?
+        if (nodeId == MYSENSORS_NODE_ID_ALL_KNOWING && childId == MYSENSORS_CHILD_ID_ALL_KNOWING) {
+            updateState(CHANNEL_MYSENSORS_MESSAGE,
+                    new StringType(MySensorsMessage.generateAPIString(msg).replaceAll("(\\r|\\n)", "")));
+
+        }
+    }
+
+    private void registerListeners() {
+        if (!myGateway.isEventListenerRegisterd(this)) {
+            logger.debug("Event listener for node {}-{} not registered yet, registering...", nodeId, childId);
+            myGateway.addEventListener(this);
+        }
+    }
+
+    private void addIntoGateway(Thing thing, MySensorsSensorConfiguration configuration) {
+        MySensorsNode node = generateNodeFromThing(thing, configuration);
+        if (node != null) {
+            myGateway.addNode(node, true);
+        } else {
+            logger.error("Failed to build sensor for thing: {}", thing.getUID());
+        }
+    }
+
+    private MySensorsNode generateNodeFromThing(Thing t, MySensorsSensorConfiguration configuration) {
+        MySensorsNode ret = null;
+        Integer nodeId = -1, childId = -1, pres = -1;
+        try {
+            nodeId = Integer.parseInt(t.getConfiguration().as(MySensorsSensorConfiguration.class).nodeId);

```

Comment:
we have a configuration as parameter, we have a configuration in this and we get the configuration from t what is going on here?